collated/main/A0139128A.md

```diff
@@ -0,0 +1,1500 @@
+				cal2.add(Calendar.DATE, 1);
+				taskDate = dateFormat2.format(cal2.getTime());
+				fullDate= taskDate;
+			}
+		}
+	}
+	/*
+	 * Validates given date
+	 *
+	 * @throw IllegalValueException if given date is invalid
+	 */
+	public TaskDate(String taskDate, String startDate, String endDate) throws IllegalValueException, java.text.ParseException {
+		ListOfDateRegex = new ArrayList<String>();
+		ListOfDateFormat = new ArrayList<String>();
+		ListOfDateRegex.add(DATE_NUM_SLASH_WITH_YEAR_VALIDATION_REGEX);
+		ListOfDateRegex.add(DATE_NUM_SLASH_WITH_YEAR_VALIDATION_SHORTENED_DAY_REGEX);
+		ListOfDateRegex.add(DATE_NUM_SLASH_WITH_YEAR_VALIDATION_SHORTENED_MONTH_REGEX);
+		ListOfDateRegex.add(DATE_NUM_SLASH_WITH_YEAR_VALIDATION_SHORTENED_DAY_AND_MONTH_REGEX);
+		ListOfDateRegex.add(DATE_NUM_SLASH_WITHOUT_YEAR_VALIDATION_REGEX);
+		ListOfDateRegex.add(DATE_ALPHA_WITH_YEAR_VALIDATION_REGEX);
+		ListOfDateRegex.add(DATE_ALPHA_WITHOUT_YEAR_VALIDATION_REGEX);
+		ListOfDateFormat.add(DATE_NUM_SLASH_WITH_YEAR_FORMAT);
+		ListOfDateFormat.add(DATE_NUM_SLASH_WITH_YEAR_SHORTENED_DAY_FORMAT);
+		ListOfDateFormat.add(DATE_NUM_SLASH_WITH_YEAR_SHORTENED_MONTH_FORMAT);
+		ListOfDateFormat.add(DATE_NUM_SLASH_WITH_YEAR_SHORTENED_DAY_AND_MONTH_FORMAT);
+		ListOfDateFormat.add(DATE_NUM_SLASH_WITHOUT_YEAR_FORMAT);
+		ListOfDateFormat.add(DATE_AlPHA_WHITESPACE_WITH_YEAR_FORMAT);
+		ListOfDateFormat.add(DATE_ALPHA_WHITESPACE_WITHOUT_YEAR_FORMAT);
+		
+		if(taskDate == null && startDate != null &&  endDate != null) {
+			if(!isValidDateRange(startDate, endDate)) {
+				throw new IllegalValueException(INVALID_TASK_DATE_RANGE_FORMAT);
+			}
+		}
+		else {
+			if(!isValidDate(taskDate)) {
+				throw new IllegalValueException(MESSAGE_NAME_CONSTRAINTS);
+			}
+			//Formats the date to be today's date
+			if(taskDate.equals("today")) {
+				DateFormat dateFormat = new SimpleDateFormat(DATE_NUM_SLASH_WITH_YEAR_FORMAT);
+				Calendar cal = Calendar.getInstance();
+				taskDate = dateFormat.format(cal.getTime());
+				fullDate = taskDate;
+			}
+			//Formats the date to be tomorrow's date
+			else if(taskDate.equals("tomorrow")) {
+				DateFormat dateFormat2 = new SimpleDateFormat(DATE_NUM_SLASH_WITH_YEAR_FORMAT);
+				Calendar cal2 = Calendar.getInstance();
+				cal2.add(Calendar.DATE, 1);
+				taskDate = dateFormat2.format(cal2.getTime());
+				fullDate = taskDate;
+			}
+		}
+	}
+
+	/**
+	 * 
+	 * @param test is a given user date input
+	 * @return the validity of the user date input by passing it to methods of different regex
+	 * @throws java.text.ParseException
+	 */
+	public boolean isValidDate(String reqDate) throws java.text.ParseException {	
+		if(reqDate.equals("today") || reqDate.equals("tomorrow")) {
+			return true;
+		}
+		else {
+			for(int i = 0 ; i < ListOfDateFormat.size() && i < ListOfDateRegex.size(); i++) {
+				if(reqDate.matches(ListOfDateRegex.get(i))) {
+					return isValidNumDate(reqDate, ListOfDateFormat.get(i));
+				}
+			}
+			return false;
```

Comment:
use 4 spaces for indentation; not tabs. you can get Eclipse to do this for you easily!